### PR TITLE
Add_SimpleCategorical

### PR DESCRIPTION
**Pull request by enkilee**
Add SimpleCategorical function 
Function: Randomly sample an integer according to the given distribution.
param1(vector<int>): candidates Candidate set of integers.
param2(vector<float>): probs Probability distribution of candidate integer set.
Return Value(int): Random variables sampled.

### CODE REVIEW

**Comment by luotao1** [comment]
还有 cinn-gpu-cudnn-off 流水线没过

**Comment by enkilee** [comment]
@luotao1 Thanks. Already done.

File: cinn/ir/ir_schedule.cc [hunk 1]
```diff
@@ -1692,6 +1693,25 @@ void ScheduleImpl::FlattenLoops(const std::vector<Expr>& loops, const bool flat_
...
+  CHECK_EQ(candidates.size(), probs.size()) << "candidates and probs must have same size.";
+  // generate
+  std::vector<double> weights;
+  for (auto p : probs) {
+    weights.push_back(p);
+  }
+  std::discrete_distribution<int32_t> prob_int(weights.begin(), weights.end());
+
+  std::random_device seed;
+  std::default_random_engine engine(seed());
```
Comment (CtfGo) [inline cinn/ir/ir_schedule.cc:1708]: 可参照SamplePerfectTile实现，在内部ScheduleImpl实现函数中新增参数传入随机数种子https://github.com/PaddlePaddle/CINN/blob/develop/cinn/ir/ir_schedule.cc#L2089

File: cinn/backends/ir_schedule_test.cc [hunk 2]
```diff
@@ -2921,5 +2921,23 @@ TEST(IrSchedule, GetChildBlocks) {
   ASSERT_EQ(utils::GetStreamCnt(ir_sch.GetChildBlocks(root_block)), expected_expr);
 }
 
+TEST(IrSchedule, SimpleCategorical) {
```
Comment (CtfGo) [inline cinn/backends/ir_schedule_test.cc:2924]: 在schedule_desc.cc添加注册SimpleCategorical Step类型，并在schedule_desc_test.cc中补充对应单测，可参照其它原语实现

**Comment by luotao1** [comment]
@enkilee 有空可以更新下 PR 哈

**Comment by enkilee** [comment]
> @enkilee 有空可以更新下 PR 哈

收到！集训回去就抓紧更新

**Pull request closed without merging**